pkg/cluster/cluster.go

```diff
@@ -55,8 +55,10 @@ type Config struct {
 
 // K8S objects that are belongs to a connection pooler
 type ConnectionPoolerObjects struct {
-	Deployment *appsv1.Deployment
-	Service    *v1.Service
+	Deployment     *appsv1.Deployment
```

Comment:
Can we define this as a map with role as a key? This will allow to avoid constant if/else.